astropy/units/scripts/typestubs_for_units.py

```diff
@@ -0,0 +1,77 @@
+"""
+Simple stub generator. Focused on `astropy.units` for now.
+
+Stub files (.pyi) help type checkers understand dynamically created code.
+
+TO DO: functions, functools._lru_cache_wrapper, method, DataInfoMeta.
+"""
+
+import astropy.units as units
+
+
+def main():
+    with open("astropy/units/__init__.pyi", "w") as f:
```

Comment:
This will only work if one is in the main astropy directory - should probably make this relative to the path to the script.

Also, if the user has no write access, this will fail (correctly). Do we need more of an error message? (I think help on the script itself is more useful.)